src/java/org/apache/cassandra/db/MultiCBuilder.java

```diff
@@ -19,104 +19,155 @@
 
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.NavigableSet;
 
+import org.apache.cassandra.cql3.statements.Bound;
+import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.schema.ColumnMetadata;
 import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.cassandra.utils.UniqueComparator;
 import org.apache.cassandra.utils.btree.BTreeSet;
 
 /**
  * Builder that allow to build multiple Clustering/ClusteringBound at the same time.
  */
-public abstract class MultiCBuilder
+public class MultiCBuilder
 {
+    /**
+     * Represents a building block of a clustering.
+     * Either a point (single value) or a bound.
+     * Knowing the kind of elements greatly simplifies the process of building clustering bounds.
+     */
+    public static class Element
```

Comment:
Nice utility class! It's not related to the patch but, since we are here, I wonder if we could/should rename the class to something more explicit, like `MultiClusteringBuilder`.

Reply:
I like `MultiClusteringBuilder` name much more than `MultiCBuilder`.

Reply:
Done